fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java

```diff
@@ -118,13 +118,19 @@ private void constructInsertStmt() {
                 // limit
                 LimitElement.NO_LIMIT
         );
+        boolean isPartialUpdate = false;
+        if (((OlapTable) targetTable).getEnableUniqueKeyMergeOnWrite()
+                && cols.size() < targetTable.getColumns().size()) {
```

Comment:
consider < 50% of total columns?
e.g. If user update 9 of total 10 columns, seems not necessary to leverage partial update.

Reply:
fixed